### DISCUSSION TITLE
Brew formula installing postgres extensions to a pinned version

### DISCUSSION THREAD

**Discussion by alexrudy** [Homebrew/discussions#1053] [Tap maintenance and Homebrew development] [answered]
I'm trying to write a fomula for postgis for use in a personal tap, but I'm running into what I think is an intentional barrier with the homebrew sandbox, and I don't know my way around formulas enough to know whether I'm barking up the wrong tree (or trying to do something impossible). I'm testing this formula on an intel Mac with macOS 11.2.3 and brew --version 3.0.5.

I'm basing the formula off of the [postgis](https://github.com/Homebrew/linuxbrew-core/blob/852d8a5305a822ac3b2a237ab084c33b6bf8be65/Formula/postgis.rb) in linuxbrew, but I want to do a few things differently, and I don't know if they are possible:

1. I want to downgrade to version 2.5.5 (seems pretty simple from the version history of the postgis formula to make a pinned version which I'm calling `postgis@2.5`
2. I want the install to make the extension available to a keg-only copy of `postgresql@11` – here is where I think I run into trouble. 
 
The formula contains a bunch of custom install lines, which seem (based on some interactive debugging) to put postgis files into folders in `/usr/local/Cellar/postgis@2.5/2.5.5/...`, for example, `/usr/local/Cellar/postgis@2.5/2.5.5/share/postgresql@11/extension/postgis.control`, when I think that Postgresql expects extensions to be installed in `/usr/local/Cellar/postgresql@11/11.11_1/share/postgresql@11/` or `/usr/local/opt/postgresql@11/share/postgresql@11/`.

I'm getting this much from running CREATE EXTENSION in Postgres and finding `could not open extension control file "/usr/local/opt/postgresql@11/share/postgresql@11/extension/postgis.control"`.

I'm not sure how to find the path `/usr/local/opt/postgresql@11/share/postgresql@11/` inside the brew install function, since it isn't part of the current formula. Is there a way to construct this path?

**Comment by carlocab** [comment, marked as answer]
You can get this path using `Formula["postgresql@11"].opt_share`.

**Reply by alexrudy** [reply]
Oh, that's really helpful! I think that almost solves my problem, but I can't install the `.so` lib files this way, because I get a permissions error when trying to do something like `Formula["postgresql@11"].opt_lib.install` – is that because the homebrew sandbox only allows writing to the `share` directory?

**Reply by carlocab** [reply]
The sandbox will not allow you to write anywhere outside of the build directory or the formula prefix at build time. You can probably write into `share/"postgresql@11"`, which will resolve into something like
```
/usr/local/Cellar/postgis@2.5/2.5.5/share/postgresql@11
```
If `postgis@2.5` is linked into the prefix, then `share/"postgresql@11"` will be linked into `/usr/local/share/postgresql@11`.
